### PR TITLE
Moved Embedder class to under neo4j_graphrag.embeddings

### PR DESCRIPTION
**Pull request by willtai**
# Description
Move the Embedder class to be under `neo4j_graphrag.embeddings` directory alongside the other custom Embedders e.g. OpenAIEmbeddings, VertexAIEmbeddings etc

## Type of Change
- [ ] New feature
- [ ] Bug fix
- [ ] Breaking change
- [ ] Documentation update
- [x] Project configuration change

## Complexity

> **Note**
>
> Please provide an estimated complexity of this PR of either Low, Medium or High
>
>

Complexity: Low

## How Has This Been Tested?
- [x] Unit tests
- [x] E2E tests
- [x] Manual tests

# Checklist

The following requirements should have been met (depending on the changes in the branch):

- [x] Documentation has been updated
- [ ] Unit tests have been updated
- [ ] E2E tests have been updated
- [x] Examples have been updated
- [ ] New files have copyright header
- [x] CLA (https://neo4j.com/developer/cla/) has been signed
- [x] CHANGELOG.md updated if appropriate

### CODE REVIEW

**Comment by willtai** [comment]
@CodiumAI-Agent /update_changelog

File: src/neo4j_graphrag/embeddings/openai.py [hunk 1]
```diff
@@ -17,7 +17,7 @@
 
 from typing import Any
 
-from neo4j_graphrag.embedder import Embedder
```
Comment (stellasia) [inline src/neo4j_graphrag/embeddings/openai.py:20]: It's maybe a detail, for the other packages (`llm` and `retrievers`), the interface (base class) is in `base.py`. Is it worth following the same sort of convention?

Comment (willtai) [reply src/neo4j_graphrag/embeddings/openai.py:20]: I was thinking that but at the same time we expect users to create custom embedders so I was thinking this is more explicit. Although following this convention could make it less confusing, wdyt?

Comment (stellasia) [reply src/neo4j_graphrag/embeddings/openai.py:20]: We also expect users to create custom retrievers, right? But maybe the 'base' convention is not a good one and we should move to something more explicit everywhere?

**Review by stellasia — APPROVED** [review]

**Pull request merged**